lib/usdUfe/ufe/UsdAttributeHolder.cpp

```diff
@@ -59,19 +201,40 @@ bool setUsdAttrMetadata(
 
     UsdUfe::AttributeEditRouterContext ctx(attr.GetPrim(), attr.GetName());
 
+    // These metadata have a USD equivalent. Use it to stay compatible with other DCCs:
+    static const auto knownUSDMetadata
+        = std::unordered_set<std::string> { UsdUfe::MetadataTokens->UIDoc,
+                                            UsdUfe::MetadataTokens->UIEnumLabels,
+                                            UsdUfe::MetadataTokens->UIName,
+                                            UsdUfe::MetadataTokens->UIFolder };
+
     PXR_NS::TfToken tok(key);
     if (PXR_NS::UsdShadeNodeGraph(attr.GetPrim())) {
         if (PXR_NS::UsdShadeInput::IsInput(attr)) {
             PXR_NS::UsdShadeInput input(attr);
-            input.SetSdrMetadataByKey(tok, value.get<std::string>());
-            return true;
+            if (knownUSDMetadata.count(key)) {
+                // Cleanup known USD metadata previously set as SdrMetadata
+                input.ClearSdrMetadataByKey(tok);
+            } else {
+                input.SetSdrMetadataByKey(tok, value.get<std::string>());
+                return true;
+            }
```

Comment:
[Suggestion] We might not necessarily need `knownUSDMetadata` anymore, since `setUsdNativeMetadata()` returns a Boolean telling us if the key was a valid USD metadata key.

So, we could possibly do:
```C++
    if (setUsdNativeMetadata(attr, key, value)) {
        // Cleanup known USD metadata previously set as SdrMetadata
        input.ClearSdrMetadataByKey(tok);
        return true;
    }
    
    input.SetSdrMetadataByKey(tok, value.get<std::string>());
    return true;
```

Not sure if having the map improves performance, but if the set of know USD metadata changes, we'd currently need to adjust both `knownUSDMetadata` and `setUsdNativeMetadata()`.

Reply:
Good idea. I put the cleanup code where it belongs so it is no longer split in two functions.